File: Codigos estudiantes por lenguaje/Java/Edwin Villarraga/matematicas/TowerHanoiRecursive.java
public class Main {

    public static void main(String[] args) {
            int n;
            n = 8;
            hanoi(n, "Primera torre", "Segunda torre", "Tercera torre");
        }
        static int paso = 1;
        static void hanoi(int n, String from, String temp, String to) {
            if (n == 0) {
                return;
            }
            hanoi(n - 1, from, to, temp);
            System.out.println(paso + " Mover disco " + n + " de " + from + " a " + to);
            paso++;
            hanoi(n - 1, temp, from, to);
        }
}

File: Codigos estudiantes por lenguaje/C++/Edwin Villarraga/matematica/TowerHanoiRecursive.cpp
#include<bits/stdc++.h>
#include<cstdlib>
using namespace std;
int paso=1;
void hanoi(int n,string from,string temp,string to){
    if(n==0){
        return;
    }
    hanoi(n-1,from,to,temp);
    cout<<paso<<" mover disco "<<n<<" de "<<from<<" a "<<to<<endl;
    paso++;
    hanoi(n-1,temp,from,to);
}
int main(){
    int n=8;
    hanoi(n,"Primer Torre","Segunda Torre","Tercer Torre");
    return 0;
}
